pkg/controller/download_request_controller_test.go

```diff
@@ -18,306 +18,249 @@ package controller
 			}
 
-			if tc.backup != nil {
-				require.NoError(t, harness.informerFactory.Velero().V1().Backups().Informer().GetStore().Add(tc.backup))
+			if test.restore != nil {
+				err := fakeClient.Create(context.TODO(), test.restore)
+				Expect(err).To(BeNil())
 			}
 
-			if tc.expectGetsURL {
-				harness.backupStore.On("GetDownloadURL", tc.downloadRequest.Spec.Target).Return("a-url", nil)
+			// Setup reconciler
+			Expect(velerov1api.AddToScheme(scheme.Scheme)).To(Succeed())
+			r := DownloadRequestReconciler{
+				Client:            fakeClient,
+				Clock:             rClock,
+				NewPluginManager:  func(logrus.FieldLogger) clientmgmt.Manager { return pluginManager },
+				BackupStoreGetter: NewFakeObjectBackupStoreGetter(backupStores),
+				Log:               velerotest.NewLogger(),
 			}
 
-			// exercise method under test
-			key := tc.key
-			if key == "" && tc.downloadRequest != nil {
-				key = kubeutil.NamespaceAndName(tc.downloadRequest)
+			if test.backupLocation != nil && test.expectGetsURL {
+				backupStores[test.backupLocation.Name].On("GetDownloadURL", test.downloadRequest.Spec.Target).Return("a-url", nil)
 			}
 
-			err := harness.controller.processDownloadRequest(key)
+			actualResult, err := r.Reconcile(context.Background(), ctrl.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: velerov1api.DefaultNamespace,
+					Name:      test.downloadRequest.Name,
+				},
+			})
 
-			// verify results
-			if tc.expectedErr != "" {
-				require.Equal(t, tc.expectedErr, err.Error())
+			Expect(actualResult).To(BeEquivalentTo(test.expectedRequeue))
+			if test.expectedReconcileErr == "" {
+				Expect(err).To(BeNil())
 			} else {
-				assert.Nil(t, err)
-			}
-
-			if tc.expectGetsURL {
-				output, err := harness.client.VeleroV1().DownloadRequests(tc.downloadRequest.Namespace).Get(context.TODO(), tc.downloadRequest.Name, metav1.GetOptions{})
-				require.NoError(t, err)
-
-				assert.Equal(t, string(velerov1api.DownloadRequestPhaseProcessed), string(output.Status.Phase))
-				assert.Equal(t, "a-url", output.Status.DownloadURL)
-				assert.True(t, velerotest.TimesAreEqual(harness.controller.clock.Now().Add(signedURLTTL), output.Status.Expiration.Time), "expiration does not match")
+				Expect(err.Error()).To(Equal(test.expectedReconcileErr))
 			}
 
-			if tc.downloadRequest != nil && tc.downloadRequest.Status.Phase == velerov1api.DownloadRequestPhaseProcessed {
-				res, err := harness.client.VeleroV1().DownloadRequests(tc.downloadRequest.Namespace).Get(context.TODO(), tc.downloadRequest.Name, metav1.GetOptions{})
+			instance := &velerov1api.DownloadRequest{}
+			err = r.Client.Get(ctx, kbclient.ObjectKey{Name: test.downloadRequest.Name, Namespace: test.downloadRequest.Namespace}, instance)
 
-				if tc.expired {
-					assert.True(t, apierrors.IsNotFound(err))
+			if test.expired {
+				if test.downloadRequest.Status.Phase == velerov1api.DownloadRequestPhaseProcessed {
```

Comment:
If `test.expired` is true, then we should always expect the err to be a `NotFound` error.

Reply:
Ah. I was thinking this test block was a mess. Thanks to your other comment, I realize a bug in the test: a request cannot be both expired and have a phase other than `processed`, and the last test case if braking this rule. Fixing this.